cmd/job/submit/submit_template_util.go

```diff
@@ -0,0 +1,160 @@
+	submitArgs.submitArgs = mergeTemplateToSubmitArgs(submitArgs.submitArgs, template, extraArgs)
+	submitArgs.BackoffLimit = mergeIntFlags(submitArgs.BackoffLimit, template.BackoffLimit)
+	submitArgs.Elastic = mergeBoolFlags(submitArgs.Elastic, template.Elastic)
+	submitArgs.Parallelism = mergeIntFlags(submitArgs.Parallelism, template.Parallelism)
+	submitArgs.IsPreemptible = mergeBoolFlags(submitArgs.IsPreemptible, template.IsPreemptible)
+	submitArgs.ServiceType = mergeStringFlags(submitArgs.ServiceType, template.ServiceType)
+	submitArgs.IsJupyter = mergeBoolFlags(submitArgs.IsJupyter, template.IsJupyter)
+	submitArgs.TtlAfterFinished = mergeDurationFlags(submitArgs.TtlAfterFinished, template.TtlAfterFinished)
+	return submitArgs
+}
+
+func mergeTemplateToMpiSubmitArgs(submitArgs submitMPIJobArgs, template *templates.SubmitTemplate, extraArgs []string) submitMPIJobArgs {
+	submitArgs.submitArgs = mergeTemplateToSubmitArgs(submitArgs.submitArgs, template, extraArgs)
+	submitArgs.Processes = mergeIntFlags(submitArgs.Processes, template.Processes)
+	return submitArgs
+}
+
+func mergeEnvironmentVariables(cliEnvVars, templateEnvVars *[]string) []string {
+	cliEnvVarMap := make(map[string]bool)
+
+	for _, cliVar := range *cliEnvVars {
+		maybeKeyVal := strings.Split(cliVar, "=")
+		if len(maybeKeyVal) != 2 {
+			continue
+		}
+		key := maybeKeyVal[0]
+		cliEnvVarMap[key] = true
+	}
+
+	for _, templateVar := range *templateEnvVars {
+		maybeKeyVal := strings.Split(templateVar, "=")
+		if len(maybeKeyVal) != 2 {
+			continue
+		}
+		key := maybeKeyVal[0]
+		if !cliEnvVarMap[key] {
+			*cliEnvVars = append(*cliEnvVars, templateVar)
+		}
+	}
+
+	return *cliEnvVars
+}
+
+func mergeBoolFlags(cliFlag, templateFlag *bool) *bool {
+	if cliFlag != nil {
+		return cliFlag
+	} else if templateFlag != nil {
+		return templateFlag
+	}
+	return nil
+}
+
+func mergeStringFlags(cliFlag, templateFlag string) string {
+	if cliFlag != "" {
+		return cliFlag
+	} else if templateFlag != "" {
+		return templateFlag
+	}
+	return ""
+}
+
+func mergeFloat64Flags(cliFlag, templateFlag *float64) *float64 {
+	if cliFlag != nil {
+		return cliFlag
+	} else if templateFlag != nil {
+		return templateFlag
+	}
+	return nil
+}
+
+func mergeIntFlags(cliFlag, templateFlag *int) *int {
+	if cliFlag != nil {
+		return cliFlag
+	} else if templateFlag != nil {
+		return templateFlag
+	}
+	return nil
+}
+
+func mergeDurationFlags(cliFlag, templateFlag *time.Duration) *time.Duration {
+	if cliFlag != nil {
+		return cliFlag
+	} else if templateFlag != nil {
+		return templateFlag
+	}
+	return nil
+}
+
+func mergeCommandAndArgs(templateIsCommand, cliIsCommand bool, templateExtraArgs, cliExtraArgs []string) ([]string, []string) {
+	if templateIsCommand && !cliIsCommand {
+		return templateExtraArgs, cliExtraArgs
+	} else if templateIsCommand && cliIsCommand {
+		return cliExtraArgs, []string{}
+	} else if !templateIsCommand && cliIsCommand {
```

Comment:
Let's merge it with the previous `if`?
```suggestion
	} else if cliIsCommand {
```